src/vs/workbench/contrib/extensions/common/extensions.ts

```diff
@@ -101,6 +101,8 @@ export interface IExtensionsWorkbenchService {
 	queryGallery(options: IQueryOptions, token: CancellationToken): Promise<IPager<IExtension>>;
 	getExtensions(extensionInfos: IExtensionInfo[], token: CancellationToken): Promise<IExtension[]>;
 	getExtensions(extensionInfos: IExtensionInfo[], options: IExtensionQueryOptions, token: CancellationToken): Promise<IExtension[]>;
+	requireReload(): Promise<IExtension[]>;
+	getReloadRequiredState(extension: IExtension, runningExtensions: Readonly<IRelaxedExtensionDescription>[]): ReloadRequiredState;
```

Comment:
I think this method can be simplified as follows:

`getReloadStatus(extension: IExtension, runningExtensions: Readonly<IRelaxedExtensionDescription>[]): string | undefined;`

I assume enablement can be computed based on the returned status value and the label will be same for all extensions require reload which is **Reload Required**